### PR TITLE
Add tooling and docs for GitLab Code Quality reporting

### LINKED ISSUE #80: Document a setup suitable for GitLab
Document a setup suitable for GitLab.

### PR DESCRIPTION
**Pull request by osslate-rh**
This PR adds the following:

- [x] A tool to convert Vale's JSON output into GitLab Code Quality JSON
- [x] End user documentation for enabling Code Quality checks for a doc repository

image

Closes #80.

### CODE REVIEW

File: modules/user-guide/partials/proc_enabling-style-compliance-reports-in-gitlab-merge-requests.adoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+[id="proc_gitlab-c_{context}"]
+= Enabling style compliance reports in GitLab merge requests
+
+[role="_abstract"]
+This procedure enables the GitLab Code Quality widget for Vale.
+The widget reports style compliance issues in the GitLab merge request interface.
+
+.Prerequisites
+
+* xref:getting-started-with-vale.adoc[]
```
Comment (themr0c) [inline modules/user-guide/partials/proc_enabling-style-compliance-reports-in-gitlab-merge-requests.adoc:12]:
```suggestion
* xref:introduction.adoc[]
```

File: modules/user-guide/pages/gitlab-code-quality-reports.adoc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+// End of metadata for Antora
+
+:parent-context-of-enabling-style-compliance-reports-in-gitlab-merge-requests: {context}
+
+[id="configuring-vale-for-gitlab-code-quality-reports_{context}"]
+= GitLab Code Quality reports
+
+:context: enabling-style-compliance-reports-in-gitlab-merge-requests
+
+include::partial$proc_enabling-style-compliance-reports-in-gitlab-merge-requests.adoc[leveloffset=+1]
```
Comment (themr0c) [inline modules/user-guide/pages/gitlab-code-quality-reports.adoc:14]: As it contains only one procedure, you could add directly the content of the procedure into the page.

File: modules/user-guide/pages/gitlab-code-quality-reports.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// Metadata for Antora
+:navtitle: GitLab Code Quality reports
```
Comment (themr0c) [inline modules/user-guide/pages/gitlab-code-quality-reports.adoc:2]: Please add `GitLab Code Quality` in `.vale/styles/RedHat/Headings.yml` exceptions (line 71):

```
  - GitLab Code Quality
```

And a test in `.vale/fixtures/RedHat/Headings/testvalid.adoc`

```

== GitLab Code Quality

```

**Comment by themr0c** [comment]
We could merge as it is as my suggestions are kitchen sink and won't change much for our audience. Do you prefer to fix things in the pull request or in a subsequent pull request?

File: modules/user-guide/pages/gitlab-code-quality-reports.adoc (hunk 2 above)
Comment (osslate-rh) [reply modules/user-guide/pages/gitlab-code-quality-reports.adoc:14]: We may need a verification procedure of some sort in the future. Will leave for now but open an issue to track it.

**Pull request merged**